dbt_common/record.py

```diff
@@ -295,9 +293,10 @@ def record_function_inner(func_to_record):
             return func_to_record
 
         @functools.wraps(func_to_record)
-        def record_replay_wrapper(*args, **kwargs):
-            recorder: Recorder = None
+        def record_replay_wrapper(*args, **kwargs) -> Any:
+            recorder: Optional[Recorder] = None
             try:
+                from dbt_common.context import get_invocation_context
```

Comment:
why move this import down?

Reply:
It avoids circular imports. I imported the Record class into dbt_common.context so that I could annotate a variable there. In the longer term I'm hoping we can avoid that reference.